docs/add_logging.md

```diff
@@ -38,3 +38,44 @@ public class MyClass
     }
 }
 ```
+
+### Additional Configuration
+
+The logging system has several configuration options that are provided to make it easier
+to use in cloud native environments.
+
+* `log.output-file` can be set to the desired output for logging and also provides a mechanism
+  to stream logs over tcp. Setting this to the format `tcp://<host>:<port>` will enable the socket
+  logger for tcp streaming.
+
+* `log.format` can be set to either `text` or `json`. When set to `json` all fields are encapsulated
+  in potentially multi-line `json` fields. This allows for capturing and indexing exceptions as singular fields in
+  a logging search system.
+
+* `log.annontation-file` allows a file containing fields to be set at startup that can be
+  interpolated from environment variables and will be included in all log output.
+
+  For example:
+  ```
+  #config.properties
+  log.annotation-file=annotations.properties
+  log.format=json
+  ```
+  ```
+  #annotations.properties
+  hostIp: ${ENV:HOST_IP}
```

Comment:
I think the `=` syntax is better here, since it's not obvious why these two properties files use a different syntax, and we always use the `=` syntax so most people wouldn't even know this is valid.